tests/helpers/test_entity_component.py

```diff
@@ -429,3 +441,53 @@ def async_loop_exception_handler(_, _2) -> None:
     assert (
         "Not passing an entity ID to a service to target all entities is deprecated"
     ) not in caplog.text
+
+
+async def test_register_entity_service(hass):
+    """Test not expanding a group."""
+    entity = MockEntity(entity_id=f"{DOMAIN}.entity")
+    calls = []
+
+    @ha.callback
+    def appender(**kwargs):
+        calls.append(kwargs)
+
+    entity.async_called_by_service = appender
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    await component.async_add_entities([entity])
+
+    component.async_register_entity_service(
+        "hello", {"some": str}, "async_called_by_service"
+    )
+
+    with pytest.raises(vol.Invalid):
+        await hass.services.async_call(
+            DOMAIN,
+            "hello",
+            {"entity_id": entity.entity_id, "invalid": "data"},
+            blocking=True,
+        )
+        assert len(calls) == 0
+
+    await hass.services.async_call(
+        DOMAIN, "hello", {"entity_id": entity.entity_id, "some": "data"}, blocking=True
+    )
+    assert len(calls) == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4946043)